### PR TITLE
Dynamic search for user, implement the skeleton for API, unit test

### PR DESCRIPTION
**Pull request by YU-ERH**
search_view/test.py now confirm result nothing since js cannot be loaded in django test. api/test.py test the /api/search/user/?query results the same thing with the UserController.searchUser

### CODE REVIEW

File: templates/search_view/search_view.html [hunk 1]
```diff
@@ -31,32 +47,61 @@ <h1>Search {{ type|title }}</h1>
...
-                {% endif %}
+                {% endfor %}
             </ul>
-        {% endfor %}
+            {% endfor %}
+        {% endif %}
     </div>
+    <script>
+        function fetchUsers(query = "") {
+            fetch(`/api/search/user/?query=${encodeURIComponent(query)}`)
```
Comment (landor341) [inline templates/search_view/search_view.html]: can we separate this into a js file in the static folder

File: views/api/test.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+from ta_scheduler.models import User
+
+class TestSearchUserAPI(TestCase):
+    def setUp(self):
+        self.client = Client()
+        User.objects.create(username='jboy', first_name='John', last_name='Boyland', password='password')
+        User.objects.create(username='lanfar', first_name='Landon', last_name='Faris', password='password')
+
+    def test_search_user_api(self):
+        response = self.client.get('/api/search/user/?query=j')
```
Comment (landor341) [inline views/api/test.py]: add a test for searching and getting no results

File: views/api/views.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    Returns:
+    - JsonResponse:
+      * On success: A JSON object with a list of users (username and name).
+      * On failure: A JSON object with an "error" key and status code 400.
+    """
+    query = request.GET.get("query", "").strip() if request.GET.get("query", "").strip() else ""
+    try:
+        users = UserController.searchUser(query)
+        user_data = [{"username": user.username, "name": user.name} for user in users]
```
Comment (landor341) [inline views/api/views.py:32]: Each user object should be a user ref with this data format so you should just be able to use users as user_data

File: views/search_view/test.py [hunk 4]
```diff
@@ -35,30 +36,30 @@ def test_post_course_search_without_semester(self):
         self.assertContains(response, "CS101")
 
 
-class TestSearchUsers(TestCase):
+class TestSearchUsersAPI(TestCase):
```
Comment (landor341) [inline views/search_view/test.py]: move these tests to the api file. And make sure to call the api, not the controller.

**Review by landor341 — CHANGES_REQUESTED** [review]
A few small comments

File: views/api/test.py (hunk 2 above)
Comment (YU-ERH) [reply views/api/test.py]: I wanna ensure api/test.py test the /api/search/user/?query results the same thing with the UserController.searchUser. What do u think

**Review by landor341 — APPROVED** [review]
awesome functionality

**Pull request merged**